### PR TITLE
test(e2e): Add local npm registry for testing

### PR DESCRIPTION
**Pull request by louis-bompart**
Based on @y-lakhdar work on #111, I restructured the test to works with https://verdaccio.org/.

Also include a shift from 'simple' docker to docker-compose: it allows to keep the docker configurations tidy and is a first step toward the better test caching.

Some modification was required in the vue specs:
Vue uses yarn, and yarn use per-package configuration, meaning the global npm registry configured was ignored.
To fix this, we have to create a .yarnrc before vue-cli actually installs stuff.
This requires new 'input hoops' in the CI (mostly, press down arrow key and enter)

Some improvement were made too to increase the overall speed of the test by removing arbitrary waits.

----
CDX-212

### CODE REVIEW

File: scripts/ui-template-utils.js
Comment (y-lakhdar) [inline scripts/ui-template-utils.js:18]: Oh. I did not think about that. Nice!

File: .github/workflows/build.yml
Comment (y-lakhdar) [inline .github/workflows/build.yml:33]: You can also remove the `NPM_TOKEN` from the env section

File: packages/cli-e2e/__tests__/angular.specs.ts
Comment (y-lakhdar) [inline packages/cli-e2e/__tests__/angular.specs.ts:17]: Did you forget to remove the skip?

File: packages/cli-e2e/__tests__/vue.specs.ts
Comment (y-lakhdar) [inline packages/cli-e2e/__tests__/vue.specs.ts:57]: I'm not sure about this for a couple of reasons:
* I think we should always overwrite the project. Merging could lead to unpredicted results
* Instead of expecting a bunch of prompts, we could simply test the default flow with the flags `--default` and `--force` using the passthrough strategy

Also, I did not include these prompts in the tests because I assumed the container running the E2E tests would always be initialized to a clean state.

File: packages/cli-e2e/__tests__/vue.specs.ts
Comment (y-lakhdar) [inline packages/cli-e2e/__tests__/vue.specs.ts:81]: Good idea. That will speed up the tests instead of adding an arbitrary timeout

File: packages/cli-e2e/__tests__/vue.specs.ts
Comment (louis-bompart) [reply packages/cli-e2e/__tests__/vue.specs.ts:57]: What's going on with the Vue tests here:
 1. We create the directory where the project
 2. We write a `.yarnrc` that tells Yarn to uses Verdaccio
 
 Given those two, if we do not merge we will discard the `.yarnrc` and thus vue-cli won't use our registry.
 
 Therefore, without changing the behavior of any command, this is the only solution I found which is 'sound'
 (I could probably also spoof npm at the DNS level but... it's a bit more involved imo)

File: packages/cli-e2e/__tests__/angular.specs.ts
Comment (louis-bompart) [reply packages/cli-e2e/__tests__/angular.specs.ts:17]: I'm stabilizing the vue test and then will stabilize the angular ones.

File: packages/cli-e2e/docker/config/config.yaml
Comment (olamothe) [inline packages/cli-e2e/docker/config/config.yaml:2]: I would cleanup the generic verdacio comments/commented out code. 

I assume you copied from their documentation ? Maybe a link would be better. 

And then, I would keep just the things that we actually use in this config, with the important/specific stuff that we need commented.

**Review by olamothe — APPROVED** [review]

**Comment by olamothe** [comment]
:gg: That's going to save us a lot of troubles !

File: packages/cli-e2e/__tests__/angular.specs.ts
Comment (louis-bompart) [reply packages/cli-e2e/__tests__/angular.specs.ts:29]: Yeah the PR is still very rough around the edges, I'm still working on cleaning this up.

- [ ] refactor openPage utils that is shared across vue, react and angular specs

File: packages/cli-e2e/__tests__/react.specs.ts
Comment (louis-bompart) [reply packages/cli-e2e/__tests__/react.specs.ts]: I guess this is an 'old' comment, 'cause I uncommented

File: packages/angular/package.json
Comment (louis-bompart) [inline packages/angular/package.json:7]: - [ ] Remove after merging #135

File: packages/cli-e2e/__tests__/vue.specs.ts
Comment (louis-bompart) [reply packages/cli-e2e/__tests__/vue.specs.ts:61]: - [ ] Double check if we can do without the local `.yarnrc`

**Review by y-lakhdar — APPROVED** [review]

**Comment by louis-bompart** [comment]
- [ ] Ensure tests fails when they fails (exit code is not correct)

**Pull request merged**
